Integrations/integration-Active_Directory_Query.yml

```
@@ -0,0 +1,1320 @@
+
+        demisto.results(demisto_entry)
+
+    ''' DATABASE OPERATIONS '''
+
+    ''' CREATE OBJECT'''
+
+    def create_user():
+        args = demisto.args()
+
+        object_classes = ["top", "person", "organizationalPerson", "user"]
+        user_dn = args.get('user-dn')
+        username = args.get("username")
+        password = args.get("password")
+        custome_attributes = args.get('custom-attributes')
+        attributes = {
+            "samAccountName": username
+        }
+
+        # set common user attributes
+        if args.get('display-name'):
+            attributes['displayName'] = args['display-name']
+        if args.get('description'):
+            attributes['description'] = args['description']
+        if args.get('email'):
+            attributes['mail'] = args['email']
+        if args.get('telephone-number'):
+            attributes['telephoneNumber'] = args['telephone-number']
+        if args.get('title'):
+            attributes['title'] = args['title']
+
+        # set user custome attributes
+        if custome_attributes:
+            try:
+                custome_attributes = json.loads(custome_attributes)
+            except Exception as e:
+                demisto.info(e.message)
+                raise Exception("Failed to parse custom attributes argument. Please see an example of this argument in the argument description.")
+            for attribute_name, attribute_value in custome_attributes.items():
+                # can run default attribute stting
+                attributes[attribute_name] = attribute_value
+
+        # add user
+        success = conn.add(user_dn, object_classes, attributes)
+        if not success:
+            raise Exception("Failed to create user")
+
+        # set user password
+        success = conn.extend.microsoft.modify_password(user_dn, password)
+        if not success:
+            raise Exception("Failed to reset user password")
+
+        # enable user and expire password
+        modification =  {
+            # enable user
+            'userAccountControl': [('MODIFY_REPLACE', NORMAL_ACCOUNT)],
+            # set to 0, to force password change on next login
+            "pwdLastSet": [('MODIFY_REPLACE', "0")]
+        }
+        modify_object(user_dn, modification)
+
+        demisto_entry = {
+            'ContentsFormat': formats['text'],
+            'Type': entryTypes['note'],
+            'Contents': "Created user with DN: {}".format(user_dn)
+        }
+        demisto.results(demisto_entry)
+
+    def create_contact():
+        args = demisto.args()
+
+        object_classes = ["top", "person", "organizationalPerson", "contact"]
+        contact_dn = args.get('contact-dn')
+
+        # set contact attributes
+        attributes = {}
+        if args.get('custom-attributes'):
+            try:
+                attributes = json.loads(args['custom-attributes'])
+            except Exception as e:
+                demisto.info(e.message)
```

```suggestion
                demisto.info(str(e))
```
`message` has been removed in python 3: https://www.python.org/dev/peps/pep-0352/#retracted-ideas